maturin support on solaris sparcv9

### Bug Description

I work on building python's cryptography package on solaris sparc. with the latest cryptography version 43.0.1 , cryptography moved from setuptool-rust to maturin for builds. 
I am unable to build maturin on solaris sparc, does it support on solaris.sparc platform? it seems latest maturin requires rustc 1.74 and newer but I am having 1.70 version. 

i tried building older version of maturin like (1.4.0 or 1.5.0 )  but build is failing for me -


error[E0609]: no field `domainname` on type `libc::utsname`
   --> /tmp/.cargo/registry/src/index.crates.io-d11c229612889eed/platform-info-2.0.2/src/platform/unix.rs:163:38
    |
163 |             equal = equal && (self.0.domainname == other.0.domainname)
    |                                      ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sysname`, `nodename`, `release`, `version`, `machine`

error[E0609]: no field `domainname` on type `libc::utsname`
   --> /tmp/.cargo/registry/src/index.crates.io-d11c229612889eed/platform-info-2.0.2/src/platform/unix.rs:163:60
    |
163 |             equal = equal && (self.0.domainname == other.0.domainname)
    |                                                            ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sysname`, `nodename`, `release`, `version`, `machine`

For more information about this error, try `rustc --explain E0609`.
error: could not compile `platform-info` (lib) due to 3 previous errors



How do i build it on sparc .Please suggest.

### Your maturin version (`maturin --version`)

1.5.0

### Your Python version (`python -V`)

3.12.6

### Your pip version (`pip -V`)

42.2

### What bindings you're using

None

### Does `cargo build` work?

- [ ] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [ ] Yes

### Steps to Reproduce

I tried -

python3 -m pip install "maturin==1.5.0" on my sparc box.

could someone please help on how to fix this issue ? 

   Compiling platform-info v2.0.3
     Running `rustc --crate-name platform_info --edition=2018 /home/user/.cargo/registry/src/index.crates.io-d11c229612889eed/platform-info-2.0.3/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=186 --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C metadata=1acdd877afd955d9 -C extra-filename=-1acdd877afd955d9 --out-dir /home/user//maturin-1.7.4/target/release/deps -C strip=debuginfo -L dependency=/home/user//maturin-1.7.4/target/release/deps --extern libc=/home/user//maturin-1.7.4/target/release/deps/liblibc-fa75efb17cae4c30.rmeta --cap-lints allow`
error[E0609]: no field `domainname` on type `utsname`
   --> /home/user/.cargo/registry/src/index.crates.io-d11c229612889eed/platform-info-2.0.3/src/platform/unix.rs:130:84
    |
130 |             debug_struct = debug_struct.field("domainname", &oss_from_cstr(&self.0.domainname));
    |                                                                                    ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sysname`, `nodename`, `release`, `version`, `machine`

error[E0609]: no field `domainname` on type `utsname`
   --> /home/user/.cargo/registry/src/index.crates.io-d11c229612889eed/platform-info-2.0.3/src/platform/unix.rs:165:38
    |
165 |             equal = equal && (self.0.domainname == other.0.domainname);
    |                                      ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sysname`, `nodename`, `release`, `version`, `machine`

error[E0609]: no field `domainname` on type `utsname`
   --> /home/user/.cargo/registry/src/index.crates.io-d11c229612889eed/platform-info-2.0.3/src/platform/unix.rs:165:60
    |
165 |             equal = equal && (self.0.domainname == other.0.domainname);
    |                                                            ^^^^^^^^^^ unknown field
    |
    = note: available fields are: `sysname`, `nodename`, `release`, `version`, `machine`

For more information about this error, try `rustc --explain E0609`.
error: could not compile `platform-info` (lib) due to 3 previous errors

Caused by:
  process didn't exit successfully: `rustc --crate-name platform_info --edition=2018 /home/user/.cargo/registry/src/index.crates.io-d11c229612889eed/platform-info-2.0.3/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=186 --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C metadata=1acdd877afd955d9 -C extra-filename=-1acdd877afd955d9 --out-dir /home/user//maturin-1.7.4/target/release/deps -C strip=debuginfo -L dependency=/home/user//maturin-1.7.4/target/release/deps --extern libc=/home/user//maturin-1.7.4/target/release/deps/liblibc-fa75efb17cae4c30.rmeta --cap-lints allow` (exit status: 1)
  
maturin version 1.7.4
rustc version 1.78.0
OS - SunOS 5.11 11.4.73.176.3 sun4v sparc

--- a/src/target.rs
+++ b/src/target.rs
@@ -85,6 +85,7 @@ pub enum Arch {
     Mipsel,
     Mips,
     Sparc64,
+    Sparcv9,
     LoongArch64,
 }
 
@@ -108,6 +109,7 @@ impl fmt::Display for Arch {
             Arch::Mipsel => write!(f, "mipsel"),
             Arch::Mips => write!(f, "mips"),
             Arch::Sparc64 => write!(f, "sparc64"),
+            Arch::Sparcv9 => write!(f, "sparcv9"),
             Arch::LoongArch64 => write!(f, "loongarch64"),
         }
     }
@@ -130,6 +132,7 @@ impl Arch {
             Arch::Mips64el | Arch::Mips64 | Arch::Mipsel | Arch::Mips => "mips",
             // sparc64 is unsupported since FreeBSD 13.0
             Arch::Sparc64 => "sparc64",
+            Arch::Sparcv9 => "sparcv9",
             Arch::Wasm32 => "wasm32",
             Arch::S390X => "s390x",
             Arch::LoongArch64 => "loongarch64",
@@ -192,7 +195,7 @@ fn get_supported_architectures(os: &Os) -> Vec<Arch> {
         Os::Dragonfly => vec![Arch::X86_64],
         Os::Illumos => vec![Arch::X86_64],
         Os::Haiku => vec![Arch::X86_64],
-        Os::Solaris => vec![Arch::X86_64, Arch::Sparc64],
+        Os::Solaris => vec![Arch::X86_64, Arch::Sparc64, Arch::Sparcv9],
         Os::Emscripten | Os::Wasi => vec![Arch::Wasm32],
         Os::Aix => vec![Arch::Powerpc64],
         Os::Hurd => vec![Arch::X86, Arch::X86_64],
@@ -288,6 +291,7 @@ impl Target {
                 _ => bail!("The architecture {} is not supported", mips32_arch),
             },
             Architecture::Sparc64 => Arch::Sparc64,
+            Architecture::Sparcv9 => Arch::Sparcv9,
             Architecture::LoongArch64 => Arch::LoongArch64,
             unsupported => bail!("The architecture {} is not supported", unsupported),
         };
@@ -370,6 +374,7 @@ impl Target {
             Arch::Mips64el | Arch::Mips64 => "mips64",
             Arch::Mipsel | Arch::Mips => "mips",
             Arch::Sparc64 => "sparc64",
+            Arch::Sparcv9 => "sparcv9",
             Arch::LoongArch64 => "loongarch64",
         }
     }
@@ -459,6 +464,7 @@ impl Target {
             | Arch::Mips
             | Arch::Powerpc
             | Arch::Sparc64
+            | Arch::Sparcv9
             | Arch::LoongArch64 => PlatformTag::Linux,
         }
     }
@@ -475,6 +481,7 @@ impl Target {
             | Arch::Mips64el
             | Arch::Mips64
             | Arch::Sparc64
+            | Arch::Sparcv9
             | Arch::LoongArch64 => 64,
             Arch::Armv6L
             | Arch::Armv7L

